File: 0007-reverse-integer/0007-reverse-integer.java
class Solution {
    public int reverse(int x) {
       if(x >= -9 && x <= 9) return x;
       else return reverseInteger(x);
 }
    public int reverseInteger(int x){
        Boolean isNeg = x<0;
        x = Math.abs(x);
        long res = 0;
        while(x>0){
            res = res*10 + x%10;
            x/=10;
        }
        if(isNeg) res = -res;
        if(res > Integer.MAX_VALUE || res < Integer.MIN_VALUE) return 0;
        return (int)res;
    }
}

File: 0007-reverse-integer/0007-reverse-integer.cpp
class Solution {
public:
    int reverse(int x) {
        int result = 0;
        int sign = (x<0)?-1:1;
        x = abs(x);
        while(x>0){
           if (result > (INT_MAX - x % 10) / 10) {
                return 0; // Return 0 in case of overflow
            }
            result = result*10 + x%10;
            x/=10;
        }
        return sign*result;
    }
};